test/rhsmlib/dbus/test_attach.py

```diff
@@ -10,187 +10,159 @@
-        self.mock_is_simple_content_access.return_value = False
-        self.addCleanup(is_simple_content_access_patcher.stop)
-
-    def injection_definitions(self, *args, **kwargs):
-        if args[0] == inj.IDENTITY:
-            return self.mock_identity
-        elif args[0] == inj.CP_PROVIDER:
-            provider = mock.Mock(spec=CPProvider, name="CPProvider").return_value
-            provider.get_consumer_auth_cp.return_value = mock.Mock(name="MockCP")
-            return provider
-        else:
-            return None
-
-    def dbus_objects(self):
-        return [AttachDBusObject]
-
-    def test_pool_attach(self):
-        def assertions(*args):
-            expected_content = [json.dumps(CONTENT_JSON), json.dumps(CONTENT_JSON)]
-            result = args[0]
-            self.assertEqual(result, expected_content)
 
-        self.mock_attach.attach_pool.return_value = CONTENT_JSON
 
-        dbus_method_args = [["x", "y"], 1, {}, ""]
-        self.dbus_request(assertions, self.interface.PoolAttach, dbus_method_args)
+class TestAttachDBusObject(DBusServerStubProvider):
+    dbus_class = AttachDBusObject
+    dbus_class_kwargs: Dict[str, Any] = {}
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        is_simple_content_access_patch = mock.patch(
+            "rhsmlib.dbus.objects.attach.is_simple_content_access",
+            name="is_simple_content_access",
+        )
+        cls.patches["is_simple_content_access"] = is_simple_content_access_patch.start()
+        cls.addClassCleanup(is_simple_content_access_patch.stop)
+
+        is_registered_patch = mock.patch(
+            "rhsmlib.dbus.base_object.BaseObject.is_registered",
+            name="is_registered",
+        )
+        cls.patches["is_registered"] = is_registered_patch.start()
+        cls.addClassCleanup(is_registered_patch.stop)
+
+        update_patch = mock.patch(
+            "subscription_manager.certlib.BaseActionInvoker.update",
+            name="update",
+        )
+        cls.patches["update"] = update_patch.start()
+        cls.addClassCleanup(update_patch.stop)
+
+        super().setUpClass()
+
+    def setUp(self) -> None:
+        self.patches["is_simple_content_access"].return_value = False
+        self.patches["is_registered"].return_value = True
+        self.patches["update"].return_value = None
+
+        AttachService_patch = mock.patch(
+            "rhsmlib.dbus.objects.attach.AttachService",
+            name="AttachService",
+            autospec=True,
+        )
+        self.mock_attach = AttachService_patch.start().return_value
+        self.addCleanup(AttachService_patch.stop)
+
+    def tearDown(self) -> None:
+        # Ensure the locale is set to English.
+        # Keeping the last defaults would sometimes halt the test suite, because
+        # sub-man would start speaking German, and tests can't handle that.
+        Locale.set("en_US.UTF-8")
```

Comment:
IMHO this is something useful to do regardless of the test, ie in `DBusServerStubProvider`; this way, there is no need to remember doing this manually in case we add a test for a non-english locale to a test class that did not have any before.

Reply:
No, it is not about obscuring, but rather make sure all the tests properly cleanup after themselves.

> We'll remember to do that if we do this, and move it if necessary.

That's **exactly** what I want to avoid, as it is too easy to add a new non-English test and forgetting about that. If there must be an English locale set after each test ends (otherwise other tests break, as you correctly wrote in the comment), then make this as a standard "teardown" step for all the tests, just like `DBusServerStubProvider` takes care of cleaning the patchings.